web-marketplace/src/ledger.tsx

```diff
@@ -96,13 +96,23 @@ export const LedgerProvider: React.FC<
   React.PropsWithChildren<{ wallet?: Wallet; walletLoaded: boolean }>
 > = ({ walletLoaded, wallet, children }) => {
   const [api, setApi] = useState<RegenApi | undefined>(undefined);
+  const [apiUpdatedWithOfflineSigner, setApiUpdatedWithOfflineSigner] =
+    useState(false);
   const [loading, setLoading] = useState<boolean>(false);
   const [error, setError] = useState<unknown>(undefined);
 
+  // Get API even if wallet is not loaded to unblock data fetching.
   useEffect(() => {
-    if (walletLoaded && !api)
+    if (!api) getApi(setApi, setLoading, setError, wallet?.offlineSigner);
+  }, [api, setApi, setLoading, setError, wallet?.offlineSigner]);
+
+  // Update API when the wallet is loaded.
+  useEffect(() => {
+    if (api && !apiUpdatedWithOfflineSigner && walletLoaded) {
```

Comment:
`walletLoaded` doesn't necessarily mean a wallet has been connected
if there's no wallet to connect to, then `walletLoaded` is set from false to true in https://github.com/regen-network/regen-web/blob/dev/web-marketplace/src/lib/wallet/hooks/useAutoConnect.tsx#L44
which means we don't need to reinstantiate the api in this case